### DISCUSSION TITLE
Ordering of data collection reporter calls

### DISCUSSION THREAD

**Discussion by divilian** [mesa/mesa#1952] [Q&A]
Are there any guarantees that when a DataCollector's `.collect()` method is called, the reporters in the `model_reporters`  (or `agent_reporters`) dictionaries will be called in any reliable order? I believe that dicts are inherently unordered, so I'm wondering if the ordering is essentially random.

**Comment by divilian** [comment]
Oh, maybe if one is using Python 3.7+, the order reporters are specified in the model_reporters dict is guaranteed to be the order in which they're called? (https://mail.python.org/pipermail/python-dev/2017-December/151283.html)

**Reply by rht** [reply]
Exactly.
